app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -49,17 +49,20 @@ def delete_resource(type, id, _data = {})
     end
 
     def options
-      return super unless params[:ems_id]
-
-      ems = ExtManagementSystem.find(params[:ems_id])
-
-      raise BadRequestError, "No CloudVolume support for - #{ems.class}" unless defined?(ems.class::CloudVolume)
-
-      klass = ems.class::CloudVolume
-
-      raise BadRequestError, "No DDF specified for - #{klass}" unless klass.respond_to?(:params_for_create)
-
-      render_options(:cloud_volumes, :form_schema => klass.params_for_create(ems))
+      if params[:id]
+        cloud_volume = resource_search(params[:id], :cloud_volumes, CloudVolume)
+        render_options(:cloud_volumes, :form_schema => cloud_volume.params_for_update)
+      elsif params[:ems_id]
+        ems = resource_search(params[:ems_id], :ext_management_systems, ExtManagementSystem)
+        raise BadRequestError, "No CloudVolume support for - #{ems.class}" unless defined?(ems.class::CloudVolume)
+
+        klass = ems.class::CloudVolume
+        raise BadRequestError, "No DDF specified for - #{klass}" unless klass.respond_to?(:params_for_create)
```

Comment:
Prefer to check for `klass.supports?(:create)` rather than checking for methods existing.

```suggestion
        raise BadRequestError, klass.unsupported_reason(:create) unless klass.supports?(:create)
```

Reply:
done